lib/Targets/AIETargetCDODirect.cpp

```diff
@@ -324,6 +324,33 @@ LogicalResult configureBdInBlock(XAie_DevInst &devInst, XAie_DmaDesc &dmaTileBd,
                             lenInBytes);
   }
 
+  // ND zero padding.
+  std::optional<llvm::ArrayRef<BDPadLayoutAttr>> padDims =
+      bdOp.getPadDimensions();
+  if (padDims) {
+    XAie_DmaPadTensor dmaPadTensor = {};
+    dmaPadTensor.NumDim = padDims->size();
+    dmaPadTensor.PadDesc = static_cast<XAie_PadDesc *>(
+        calloc(dmaPadTensor.NumDim, sizeof(XAie_PadDesc)));
+    if (!dmaPadTensor.PadDesc)
+      return bdOp.emitError("couldn't allocate array of XAie_PadDesc");
+    // libxaie requires stride in multiples of 32b
+    double elementWidthIn32bWords =
+        static_cast<double>(bdOp.getBufferElementTypeWidthInBytes()) / 4.0;
+    for (size_t i = 0; i < padDims->size(); i++) {
+      // Pass down dimensions in reverse order.
+      int j = padDims->size() - i - 1;
+      uint8_t before;
+      uint8_t after;
+      before = static_cast<uint8_t>(padDims.value()[i].getZeroPadBefore() *
```

Comment:
Can the padding actually be less than one data beat?  Does it work to pad by one byte?

Reply:
No. In the underlying hw it is counting as either (1) if innermost dimension or if zero padding before, # 32-bit words, or else (2) number of n-1 wraps.

This kind of things really need to be enforced with proper verifiers.

Reply:
Verifier updated.